app/controllers/unattended_controller.rb

```diff
@@ -79,7 +79,12 @@ def get_host_details
     @host = find_host_by_spoof || find_host_by_token || find_host_by_ip_or_mac
     unless @host
       logger.info "#{controller_name}: unable to find a host that matches the request from #{request.env['REMOTE_ADDR']}"
-      head(:not_found) and return
+      if @spoof
+        render(:text => _("Host not found or unauthorized to view this host"), :status => :not_found)
+      else
+        head(:not_found)
+      end
+      return
     end
     unless @host.operatingsystem
       logger.error "#{controller_name}: #{@host.name}'s operating system is missing!"
```

Comment:
The other errors in this method should be handled in the same way as above, and to be honest you could simply render the same :text string whether it's a spoof or not which would simplify the code.